### ISSUE TITLE
Q about metric: EPE and Bad 1.0 

### ISSUE THREAD

**Issue by TwiceMao** [princeton-vl/RAFT-Stereo#39]
I think the definition formula of EPE is as follows:
![epe](https://user-images.githubusercontent.com/43490149/161006620-d7b46aad-4071-4d1e-bf4e-067b7849d765.png)
Am I right？
Although I have read the codes evaluate_stereo.py. , I am confused with the definition of bad pixel 1.0.
Could you help me understand it. Wish for your early reply, thank you!

**Comment by lahavlipson** [comment]
EPE is just the euclidean distance between the predicted flow/disparity vector and the GT flow/disparity vector. Typically methods report the average EPE over the entire image. The equation above seems to show something different.

bad pixel 1.0 is the percent of EPE values that are greater than 1.0.

**Comment by TwiceMao** [comment]
OK，your answer is important to me, thanks. 
And I also wondering about the codes of evaluate_stereo.py corresponding to evaluating EPE and D1.
I think there is no need to assign parameter "dim = 0" in the 85th line of evaluate_stereo.py. 
Why not just  "epe = torch.sum((flow_pr - flow_gt)**2).sqrt()“” to get a whole sum?
And ">= 0.5" in the 89th line and "epe_flattened > 3.0" in the 91th line both make me confused.
I can not match meaning with those codes.
Wish for your early reply, thank you!
![evaluate_stereo](https://user-images.githubusercontent.com/43490149/161370290-1ae27c59-e48b-4ba3-a63b-62ca8c19a5b5.png)

**Comment by lahavlipson** [comment]
You are correct, the `dim = 0` is not needed. This evaluation code is meant to also work for 2D flow.

The `>= 0.5` is to convert an array of binary ints to an array of bool. The `> 3.0` is to calculate the bad 3.0 error.

**Comment by TwiceMao** [comment]
What does you mean by "The >= 0.5 is to convert an array of binary ints to an array of bool. "? 
I can not understand.
Could you explain it with an example?

**Comment by lahavlipson** [comment]
```
>>> torch.tensor([1,0,1]) >= 0.5
tensor([ True, False,  True])
```

**Comment by TwiceMao** [comment]
But why "0.5" ?
I mean what's the meaning of the numer "0.5" ?
I do not think it is corresponding to bad 0.5.

**Comment by sjhdhs** [comment]
Hello, one element of epe-list is nan, which causes epe(things-epe) to be nan. I wonder what is the reason
![捕获](https://user-images.githubusercontent.com/84086169/220856324-38b5013e-c9fc-472a-9149-b50ff3ab4256.PNG)

**Comment by Bryan-Rathos** [comment]
@sjhdhs I noticed this too. It looks like none of the pixels in one particular image are valid based on the condition you have highlighted. Did you dig deeper into this? A simple solution would be to ignore the NaN and not add it to the pe_list. However, I wonder how it worked for the authors.
